tests/UtopiaRequestTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Utopia\Tests;
+
+use Utopia\Request as UtopiaRequest;
+
+class UtopiaRequestTest extends UtopiaRequest
+{
+    /**
+     * @var array
+     */
+    private static $params = null;
+
+    /**
+     * Get Param
+     *
+     * Get param by current method name
+     *
+     * @param string $key
+     * @param mixed $default
+     * @return mixed
+     */
+    public function getParam(string $key, $default = null): mixed
+    {
+        if($this::_hasParams() && \in_array($key, $this::_getParams())) {
+            return $this::_getParams()[$key];
+        }
+
+        return parent::getParam($key, $default);
+    }
+
+    /**
+     * Get Params
+     *
+     * Get all params of current method
+     *
+     * @return array
+     */
+    public function getParams(): array
+    {
+        $paramsArray = [];
+
+        if($this::_hasParams()) {
+            $paramsArray = $this::_getParams();
+        }
+
+        return \array_merge($paramsArray, parent::getParams());
+    }
+
+
```

Comment:
If you are extending the request class only to add a `setParams` function, we only need to override the `setParams` method. Is there a reason you have implemented all the other methods as well ?